### DISCUSSION TITLE
Is it safe to use regions as locales?

### DISCUSSION THREAD

**Discussion by nring** [amannn/next-intl#1251] [Q&A]
I'm setting up a large content-driven public-facing website and am facing an issue with regions and locales. The team won't have enough resources to enter translations for many locales with the exception of fr-CA. However their content will differ between regions. The default for all regions would be english. Regions would likely have additional variations for spelling ("color" vs "colour"). So consider the following examples:

`/` - Homepage for en-US
`/canada` - Homepage for en-CA
`/canada-fr` - Homepage for fr-CA
`/apac` - Homepage for visitors from anywhere in Asia/Pacific
`/europe` - Homepage for visitors from anywhere in Europe
`/[region]/common-page` - Common page available to all regions with appropriately translated content

Additionally, not all content like the homepage will be available to all regions:

`/blog/us-specific-post` - Available if the user is routed to the default region.
`/apac/blog/japan-specific-post` - Only available if the user is routed to the Asia/Pacific Region
`/apac/blog/us-specific-post` - 404

I'm planning on having a region switcher component to allow the user to manually select their region and language. But I was thinking about using an IP location detection service to determine what region to initially route the user rather than try to map all the different locales to a specific region. 

Using `next-intl`, would it be safe to treat each of my regions as a locale? Or would that get me into trouble the further my app progresses. I have seen a similar discussion and feature in #880, but I think this is a bit different since I have no intention of using both `[region]` and `[locale]` in my routes.

**Comment by amannn** [comment]
Have you seen the [custom prefixes](https://next-intl-docs.vercel.app/docs/routing#locale-prefix-custom) docs? It's certainly an option to use e.g. `/canada` as a prefix for the `en-CA` locale.

Partially translated routes is currently being tracked in https://github.com/amannn/next-intl/issues/990, this might require some work on your side currently (e.g. handling 404s for unavailable language variants, possibly adapting alternate links, …).

Hope this helps to get off the ground, let me know how it goes!
